hoot-core/src/main/cpp/hoot/core/io/OsmMapWriterFactory.cpp

```diff
@@ -127,10 +127,17 @@ bool OsmMapWriterFactory::hasElementOutputStream(QString url)
 
 void OsmMapWriterFactory::write(const boost::shared_ptr<const OsmMap> &map, QString url)
 {
-  LOG_INFO("Writing map to " << url << "...");
-  boost::shared_ptr<OsmMapWriter> writer = getInstance().createWriter(url);
-  writer->open(url);
-  writer->write(map);
+  if (map->isEmpty() && ConfigOptions().getOsmMapWriterSkipEmptyMap())
+  {
+    LOG_INFO("Map is empty. Not writing to " << url << "...");
+  }
+  else
+  {
+    LOG_INFO("Writing map to " << url << "...");
+    boost::shared_ptr<OsmMapWriter> writer = getInstance().createWriter(url);
+    writer->open(url);
```

Comment:
The more I think about it, the more I'd like to see a code construct like:
`open`
`write`
`close`

rather than having the `write` call sneakily close things under the covers. I think this would make the code more maintainable.

Reply:
Done.
You now have a few more files to review :-)

Reply:
Undone :-(

I added a close method to OsmJsonWriter and am using it instead of _fp.close();

BUT, I now have a failing test:  splitLongWays

Reply:
Ref: open->write->close.
From OsmMapWriter.h 
```
/**
   * Writes the specified map out. When this method is complete the output will likely be closed
   * and all data is guaranteed to be flushed.
   */
  virtual void write(ConstOsmMapPtr map) = 0;
```